onnxruntime/core/platform/linux/device_discovery.cc

```diff
@@ -114,6 +114,18 @@ std::optional<bool> IsGpuDiscrete(uint16_t vendor_id, uint16_t device_id) {
   return std::nullopt;
 }
 
+Status GetPciBusId(const std::filesystem::path& sysfs_path, std::string& pci_bus_id) {
+  std::error_code error_code;
+  auto pci_bus_id_path = std::filesystem::read_symlink(sysfs_path / "device", error_code);  // e.g. 0000:65:00.0
```

Comment:
> // e.g. 0000:65:00.0

would it be worth validating that the path filename matches the expected format?